pkg/oc/admin/migrate/scc/scc.go

```diff
@@ -0,0 +1,281 @@
+		Converts SecurityContextConstraints to the equivalent PodSecurityPolicy objects.
+		For each SCC it also creates a ClusterRole with the similar name and ClusterRoleBinding
+		for every user or group that SCC has. To easy identify such auto-generated roles and bindings,
+		their names are prefixed by 'psp:' string.
+
+		NOTE: it does not modify anything by default.`)
+
+	internalMigrateSCCExample = templates.Examples(`
+		# Perform a dry-run of migrating all objects
+		%[1]s
+
+		# To actually perform the migration, the confirm flag must be appended
+		%[1]s --confirm`)
+)
+
+type migrateSCCOptions struct {
+	kubeClient *kubernetes.Clientset
+	migrate.ResourceOptions
+}
+
+func NewCmdMigrateSCC(name, fullName string, f *clientcmd.Factory, in io.Reader, out, errout io.Writer) *cobra.Command {
+	options := &migrateSCCOptions{
+		ResourceOptions: migrate.ResourceOptions{
+			In:     in,
+			Out:    out,
+			ErrOut: errout,
+
+			AllNamespaces: true,
+			Include:       []string{"securitycontextconstraints.security.openshift.io"},
+		},
+	}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   internalMigrateSCCShort,
+		Long:    internalMigrateSCCLong,
+		Example: fmt.Sprintf(internalMigrateSCCExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(options.Complete(name, f, cmd, args))
+			kcmdutil.CheckErr(options.Validate())
+			kcmdutil.CheckErr(options.Run(cmd, out))
+		},
+	}
+	options.ResourceOptions.Bind(cmd)
+
+	// these flags aren't applied for SCC migrations
+	cmd.Flags().MarkHidden("to-key")
+	cmd.Flags().MarkHidden("from-key")
+	cmd.Flags().MarkHidden("include")
+	cmd.Flags().MarkHidden("filename")
+	cmd.Flags().MarkHidden("show-all")
+	cmd.Flags().MarkHidden("no-headers")
+	cmd.Flags().MarkHidden("show-labels")
+	cmd.Flags().MarkHidden("all-namespaces")
+
+	return cmd
+}
+
+func (o *migrateSCCOptions) Complete(name string, f *clientcmd.Factory, c *cobra.Command, args []string) error {
+	o.ResourceOptions.SaveFn = o.save
+	if err := o.ResourceOptions.Complete(f, c); err != nil {
+		return err
+	}
+	kubeClient, err := f.KubernetesClientSet()
+	if err != nil {
+		return err
+	}
+	o.kubeClient = kubeClient
+	return nil
+}
+
+func (o *migrateSCCOptions) Validate() error {
+	if o.kubeClient == nil {
+		return errors.New("Kubernetes client is required")
+	}
+	if len(o.ResourceOptions.Include) != 1 || o.ResourceOptions.Include[0] != "securitycontextconstraints.security.openshift.io" {
+		return fmt.Errorf("the only supported resources are securitycontextconstraints")
+	}
+	return o.ResourceOptions.Validate()
+}
+
+func (o *migrateSCCOptions) Run(cmd *cobra.Command, out io.Writer) error {
+	return o.ResourceOptions.Visitor().Visit(func(info *resource.Info) (migrate.Reporter, error) {
+		_, wasSCC := info.Object.(*securityapi.SecurityContextConstraints)
+		if !wasSCC {
+			return nil, fmt.Errorf("unrecognized object %#v", info.Object)
+		}
+
+		// all SCCs have to be migrated
```

Comment:
@enj What are the assumptions for migrate commands? Should I add some filtering here to not process already processed SCCs? Or this command won't run twice?